drivers/usbhid-ups.c

```diff
@@ -114,6 +114,13 @@ bool_t use_interrupt_pipe = FALSE;
 #endif
 static time_t lastpoll; /* Timestamp the last polling */
 hid_dev_handle_t udev;
+/**
+ * CyberPower UT series sometime need a bit of help deciding their online status.
+ * This quirk is to enable the special handling of OL & DISCHRG at the same time
+ * as being OB (on battery power/no mains power)
+ */
+#define DEFAULT_CPUTQUIRK 0
+static int cputquirk = DEFAULT_CPUTQUIRK;
```

Comment:
I do not like the name, primarily as it may be misunderstood as some CPU-related quirk: "What is a CPU T? Temperature?"

And also it is a sort of finger-pointing to a "person" of vendor/model rather than the action this quirk does (and which might apply to other vendors/models eventually, if the problem proves to be more widespread it could become a more global option across drivers, potentially).

Thinking of it, maybe not a boolean but something descriptively named and extensible like `onlinedischarge="[OL|OB]"` might fit better?

Reply:
I changed the name, but I'm not exactly sure about how to use the framework to make a proper named option. Could you point me at an example somewhere? Thanks.